tests/test_dfa.py

```diff
@@ -1308,3 +1308,162 @@ def test_show_diagram_write_file(self):
         self.dfa.show_diagram(path=diagram_path)
         self.assertTrue(os.path.exists(diagram_path))
         os.remove(diagram_path)
+
```

Comment:
Maybe not totally needed, but I merged a function that creates a minimal DFA accepting a finite language, so a cool test would be to feed in a finite language to that and check that your code here is able to recover it.

Reply:
Agreed

Reply:
I added a test for iter that does this